src/runtime/thread_pool.cc

```diff
@@ -307,6 +289,14 @@ class ThreadPool {
     return res;
   }
 
+  void Shutdown() {
+    for (std::unique_ptr<SpscTaskQueue>& q : queues_) {
```

Comment:
Shutdown need to be idempotent(it is safe to make multiple calls into it) so need to check num_workers_, or cleanup the queues, and set threads_ to nullptr

Reply:
do unique_ptr.reset

Reply:
need to clear the queues_

Reply:
The queues can't be cleared on the SGX side since the untrusted world needs to join the threads. Instead, I made it so that further events can't be Enqueued.